client/src/resources/transaction.rs

```diff
@@ -162,6 +204,55 @@ mod transaction_tests {
              gAC1GEAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAAAAAAAAAAAEA/5qzAAAAAAAAAAB9h5rdSVjH6gHVJQ3\
              slLTvVfMz6idGixvCq2cl7+/EBgAqs9CXAFM+AAiBvgAC1GIAAAAAAAAAAAAAAAAAAAAAAQAAAAAAAAAAAAAAA\
              AAAAA=="
-        )
+        );
+        if let &Memo::Text(ref value) = transaction.memo() {
+            assert_eq!(value, "hello");
+        }
+    }
+
+    #[test]
+    fn it_parses_memo_text() {
+        let transaction: Transaction = serde_json::from_str(&transaction_json_memo_text()).unwrap();
+        if let &Memo::Text(ref value) = transaction.memo() {
+            assert_eq!(value, "hello");
+        }
+    }
+
+    #[test]
+    fn it_parses_memo_hash() {
+        let transaction: Transaction = serde_json::from_str(&transaction_json_memo_hash()).unwrap();
+        if let &Memo::Hash(ref value) = transaction.memo() {
+            assert_eq!(
+                value,
+                "2CF24DBA5FB0A30E26E83B2AC5B9E29E1B161E5C1FA7425E73043362938B9824"
+            );
+        }
+    }
+
+    #[test]
+    fn it_parses_memo_id() {
+        let transaction: Transaction = serde_json::from_str(&transaction_json_memo_id()).unwrap();
+        if let &Memo::Id(value) = transaction.memo() {
+            assert_eq!(value, 19292920);
+        }
+    }
+
+    #[test]
+    fn it_parses_memo_return() {
+        let transaction: Transaction =
+            serde_json::from_str(&transaction_json_memo_return()).unwrap();
+
+        if let &Memo::Return(ref value) = transaction.memo() {
```

Comment:
testing tip: When you use a conditional that should be true, you might consider putting a panic in the `else` branch so that if a different type is returned it fails.

Reply:
Awesome yes, if the if-let case fails the test will still pass. Code updated!